docs/design/2022-06-15-phyical_plan_explain.md

```diff
@@ -0,0 +1,451 @@
+
+#### `IndeRegionTaskExecxScan`
+
+In TiSpark, an `IndexScan` requires two scans. The first scan is scanning in index data and the second scan is scanning in table data to get the data we need from the results returned in the first scan.
+
+#### `CoveringIndexScan`
+
+`CoveringIndexScan` is a special case of `IndexScan`. If the column(s) in `Filter` and the column(s) in `Projection` are both inside the Index, then we only need to scan the index once, no need to scan for the table. That is, it only scans for the index, such a scan is called `CoveringIndexScan`.
+
+#### `TableScan`
+
+`TableScan` is different from `IndexScan` and `CoveringIndexScan`. `TableScan` only scans table data.
+
+### Meaning of `Filter`
+
+#### `Downgrade Filter`
+
+If the second scan in `IndexScan` does too many queries on COP([TiKV Coprocessor](https://docs.pingcap.com/tidb/stable/tikv-overview#tikv-coprocessor))/TiKV, it can cause performance problems in COP/TiKV. To solve this problem a downgrading mechanism is introduced. `Downgrade Filter` is used after `IndexScan` downgrade.
+
+The first scan of `IndexScan` will return the data that meets the conditions, and then TiSpark will sort and aggregate the returned data to obtain the `regionTask` that needs to be done in the `IndexScan` second scan. If the number of `regionTask` is bigger than `downgradeThreshold`, a downgrade will be triggered. When a downgrade is triggered, the range of the second scan table will be changed from the returned value of the first scan index to all values in the middle of the minimum to maximum value of the first scan index, and the `filters` of the second scan will change to `downgradeFilters`(`downgradeFilters` is the same as if the execution plan is `TableScan`'s `filters`).
+
+> **`RegionTask`**
+>
+> For all returned data, all consecutive data in a region will be treated as a `regionTask`.
+>
+> For example like this the data returned in the first stage are 1, 3, 4, 5 and 1, 3, 4 are in the same region and 5 is in another region. Since 1 and 3, 4 are not contiguous, 1 is a `regionTask`, and since 3, 4 and 5 are not in a region, 3, 4 is a `regionTask` and 5 is a `regionTask`. The `regionTask` number is three.
+
+#### `Residual Filter`
+
+In the original design, the `Residual Filter` represents operators that cannot be pushed down to COP/TiKV. However, in the current implementation, before the construction of `TiDAGRequest`, it will judge whether the operators can be pushed down, and only the operators that can be pushed down will participate in the construction of `TiDAGRequest`. This means that all operators in `TiDAGRequest` can be pushed down to COP/TiKV. The `Residual Filter` loses its original meaning because there are no operators in `TiDAGRequest` that cannot be pushed down.
+
+#### `PushDown Filter`
+
+The expression passed to COP/TiKV as a selection expression without triggering a downgrade.
+
+### The Problem of DAG Explain
+
+1. appear a filter that should not appear
```

Comment:
Since it's all about `Residual Filter`, why not just explicitly indicate `Residual Filter`

Reply:
changed